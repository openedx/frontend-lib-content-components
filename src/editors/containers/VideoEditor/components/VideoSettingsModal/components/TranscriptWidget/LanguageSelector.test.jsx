import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import { LanguageSelector } from './LanguageSelector';
import { formatMessage } from '../../../../../../../testUtils';

const lang1 = 'kLinGon';
const lang1Code = 'kl';
const lang2 = 'eLvIsh';
const lang2Code = 'el';
const lang3 = 'sImLisH';
const lang3Code = 'sl';

jest.mock('../../../../../../data/constants/video', () => ({
  videoTranscriptLanguages: {
    [lang1Code]: lang1,
    [lang2Code]: lang2,
    [lang3Code]: lang3,
  },
}));

describe('LanguageSelector', () => {
  const props = {
    intl: { formatMessage },
    onSelect: jest.fn().mockName('props.OnSelect'),
    title: 'tITle',
    language: lang1Code,
    openLanguages: [[lang2Code, lang2], [lang3Code, lang3]],

  };
  describe('snapshot', () => {
    test('transcript option', () => {
      expect(
        shallow(<LanguageSelector {...props} />).snapshot,
      ).toMatchSnapshot();
    });
  });
  describe('snapshots -- no', () => {
    test('transcripts no Open Languages, all should be disabled', () => {
      expect(
        shallow(<LanguageSelector {...props} openLanguages={[]} />).snapshot,
      ).toMatchSnapshot();
    });
  });
});
