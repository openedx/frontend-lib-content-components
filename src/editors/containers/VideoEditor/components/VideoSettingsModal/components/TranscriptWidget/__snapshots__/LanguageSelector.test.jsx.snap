// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector snapshot transcript option 1`] = `
<Fragment>
  <Dropdown
    className="w-100 mb-2"
  >
    <Dropdown.Toggle
      aria-label="Languages"
      block={true}
      className="w-100"
      iconAs="Button"
      id="selectLanguage-form-undefined"
      variant="outline-primary"
    >
      <ActionRow>
        kLinGon
        <ActionRow.Spacer />
        <Icon
          className="text-primary-500"
        />
      </ActionRow>
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item>
        kLinGon
        <Icon
          className="text-primary-500"
        />
      </Dropdown.Item>
      <Dropdown.Item
        onClick={[Function]}
      >
        eLvIsh
      </Dropdown.Item>
      <Dropdown.Item
        onClick={[Function]}
      >
        sImLisH
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
  <FileInput
    acceptedFiles=".srt"
    fileInput={
      {
        "addFile": [Function],
        "click": [Function],
        "ref": {
          "current": undefined,
        },
      }
    }
  />
</Fragment>
`;

exports[`LanguageSelector snapshots -- no transcripts no Open Languages, all should be disabled 1`] = `
<Fragment>
  <Dropdown
    className="w-100 mb-2"
  >
    <Dropdown.Toggle
      aria-label="Languages"
      block={true}
      className="w-100"
      iconAs="Button"
      id="selectLanguage-form-undefined"
      variant="outline-primary"
    >
      <ActionRow>
        kLinGon
        <ActionRow.Spacer />
        <Icon
          className="text-primary-500"
        />
      </ActionRow>
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item>
        kLinGon
        <Icon
          className="text-primary-500"
        />
      </Dropdown.Item>
      <Dropdown.Item
        className="disabled"
      >
        eLvIsh
      </Dropdown.Item>
      <Dropdown.Item
        className="disabled"
      >
        sImLisH
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
  <FileInput
    acceptedFiles=".srt"
    fileInput={
      {
        "addFile": [Function],
        "click": [Function],
        "ref": {
          "current": undefined,
        },
      }
    }
  />
</Fragment>
`;
